wavefront_pyformance/wavefront_reporter.py

```diff
@@ -27,26 +28,41 @@ def __init__(self, source='wavefront-pyformance', registry=None,
         self.prefix = prefix
         self.tags = tags or {}
 
+    @staticmethod
+    def decode_key(key):
+        """Decode encoded key into original key and dict of tags."""
+        if '-tags=' in key:
+            decoded_str = key.split('-tags=')
+            return decoded_str[0], json.loads(decoded_str[1])
+        return key, None
+
     def report_now(self, registry=None, timestamp=None):
         """Collect metrics from registry and report them to Wavefront."""
         registry = registry or self.registry
         metrics = registry.dump_metrics()
         for key in metrics.keys():
             is_delta = delta.is_delta_counter(key, registry)
             for value_key in metrics[key].keys():
+                metric_name, metric_tags = self.decode_key(key)
+                tags = {}
```

Comment:
To minimize allocations, can we do this only if metric_tags is non-empty:

tags = self.tags
if metric_tags:
    tags = {}
    tags.update(self.tags)
    tags.update(metric_tags)